Populate test db for loading-and-dumping to h2 with all types of entities

kidd opened the pull request:
Populating a bigger db for load-dump-load tests.  Having at least a row of each relevant entities gives more relevance to this test.

Notes:
1. I found that there's `mt/with-temp*` that does like my `insert-with-overrides` and probably more and better. The difference is that it's a "wrapping" macro.  Should I change to using it?
2. I've added new default values for some Entities. Should activity default to having a user or not?

kidd committed:
Add test for load from h2 and dump to h2

kidd committed:
Move load-and-dump-test to metabase.cmd ns

kidd committed:
sort requires

kidd committed:
delete test-load-and-dump.sh

The same test is done inside the main test suite in
`test/metabase/cmd/load_and_dump_test.clj`

kidd committed:
Fix test on h2 db

When testing with h2 driver as the destination db, special case the
jdbc spec.

H2 being in memory makes that the usual jdbc connection that would be
created would be ephemeral, so when doing the dump, and comparing
again against the first, the db would be gone. Adding a CLOSE_DELAY=10
makes it stick for a little while so we can do the comparison.

not calling `tx/create-db!` in that case because in h2, db is always
created already.

kidd committed:
Populate h2 db with all kinds of entities

kidd committed:
Add activity, label, and card-label to the populated test db

kidd committed:
Add remaining entities

Create `insert-with-overrides` helper to reduce boilerplate a bit.

kidd committed:
fix(test) add explicit nil on user_id

Activity default values now include a user

camsaul commented:
`mt/with-temp` creates new objects and then deletes them at the end of the body... tests use those to create a bunch of objects and then clean up after themselves when they're done. So it probably wouldn't work for purposes of loading up a DB with a ton of data and persisting that data

camsaul commented:
> Should activity default to having a user or not?

For the purposes of coming up with a bunch of data for this you might want to consider generating both Activities that have users and ones that don't. I'd try to make these tests cover the widest possible spectrum of objects in the DB possible. `with-temp-defaults` is a good starting point but those don't really give you random objects at all... it might make sense to come up with a more general random object protocol where you can do something like create a random dashboard and it will create a random number of cards and other stuff for it as well. This would be good for doing a more generative style of tests elsewhere as well

kidd committed:
add generator of entities for tests

kidd committed:
remove unused insert-with-overrides

kidd committed:
use a temporary db file to populate

The db fixture is checked into the repo.
Previous to this commit, running this test leaves a dirty db file,
prone to error of commiting it.

kidd committed:
clean test
